custom_fsync/sync.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+func main() {
+	if len(os.Args) != 1 {
+		fmt.Fprintf(os.Stderr, "Usage: %s\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	files, err := os.ReadDir(".")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading directory: %v\n", err)
+		os.Exit(1)
+	}
+
+	for _, file := range files {
+		filePath := file.Name()
+		if file.IsDir() {
+			continue
+		}
+
+		f, err := os.OpenFile(filePath, os.O_RDONLY, 0)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error opening file: %s: %v\n", filePath, err)
+			continue
+		}
+		defer f.Close()
+
+		if err := f.Sync(); err != nil {
+			fmt.Fprintf(os.Stderr, "Error fsyncing file: %s: %v\n", filePath, err)
+			continue
```

Comment:
for my understanding, why does this work when the previous implementation didn't?

Reply:
- OFS will only hard flush files to the mount when it gets an `fsync` syscall on a specific file descriptor
- calling just `sync` without path argument doesn't call `fsync` on the file descriptor, so file isn't hard synced
- calling `sync some_file_path` with `fsync` `some_file_path`. However some slim distros don't support calling sync with arguments, so sometimes this fails.
- i tested this solution on our fuse, and we get an `fsync` syscall for every file in the directory, so OFS should too since they also use FUSE.